### ISSUE TITLE
Feature Request: Check for duplicate methods

### ISSUE THREAD

**Issue by Azureblade3808** [microsoft/pylance-release#865]
```python
class C:
    def f(self):
        return 0

    def f(self):
        return 1
```
The code above is obviously problematic, but Pylance doesn't emit any warnings as for now (version 2021.1.2).

It usually happens after a code merge, and if the duplicate methods lie somewhere far apart, it could be hard for people to realize. It would be very helpful if Pylance could find these duplicate methods and mark them out.

**Comment by erictraut** [comment]
Thanks for the suggestion. Most of the logic was already in place to detect duplicate symbol declarations (e.g. a function that is overridden by a variable or vice versa), but it was disabled when checking for functions that override other functions. Those are a little more subtle to handle because of `@overload` and property setters/deleters. I've added the extra logic to handle those, and this new functionality will be in the next version of pylance.

Incidentally, this functionality identified six real bugs in my team's Python source base, so this was definitely a good suggestion!

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.1.3, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202113-27-january-2021
